src/components/Sections/GoogleMapsSection/index.tsx

```diff
@@ -0,0 +1,251 @@
+    stylers: [{ color: "#f5f5f5" }, { lightness: 20 }],
+  },
+  {
+    featureType: "road.highway",
+    elementType: "geometry.fill",
+    stylers: [{ color: "#ffffff" }, { lightness: 17 }],
+  },
+  {
+    featureType: "road.highway",
+    elementType: "geometry.stroke",
+    stylers: [{ color: "#ffffff" }, { lightness: 29 }, { weight: 0.2 }],
+  },
+  {
+    featureType: "road.arterial",
+    elementType: "geometry",
+    stylers: [{ color: "#ffffff" }, { lightness: 18 }],
+  },
+  {
+    featureType: "road.local",
+    elementType: "geometry",
+    stylers: [{ color: "#ffffff" }, { lightness: 16 }],
+  },
+  {
+    featureType: "poi",
+    elementType: "geometry",
+    stylers: [{ color: "#f5f5f5" }, { lightness: 21 }],
+  },
+  {
+    featureType: "poi.park",
+    elementType: "geometry",
+    stylers: [{ color: "#dedede" }, { lightness: 21 }],
+  },
+  {
+    elementType: "labels.text.stroke",
+    stylers: [{ visibility: "on" }, { color: "#ffffff" }, { lightness: 16 }],
+  },
+  {
+    elementType: "labels.text.fill",
+    stylers: [{ saturation: 36 }, { color: "#333333" }, { lightness: 40 }],
+  },
+  { elementType: "labels.icon", stylers: [{ visibility: "off" }] },
+  {
+    featureType: "transit",
+    elementType: "geometry",
+    stylers: [{ color: "#f2f2f2" }, { lightness: 19 }],
+  },
+  {
+    featureType: "administrative",
+    elementType: "geometry.fill",
+    stylers: [{ color: "#fefefe" }, { lightness: 20 }],
+  },
+  {
+    featureType: "administrative",
+    elementType: "geometry.stroke",
+    stylers: [{ color: "#fefefe" }, { lightness: 17 }, { weight: 1.2 }],
+  },
+] as google.maps.MapTypeStyle[];
+
+@Component({
+  name: "GoogleMapsSection",
+})
+class GoogleMapsSection extends BaseComponent<GoogleMapsSectionProps> {
+  map: google.maps.Map | null = null;
+  selectedIndex: number | null = null;
+  markers: MarkerWithInfoWindow[] | null = null;
+
+  @Prop() title: GoogleMapsSectionProps["title"];
+  @Prop({ required: true }) apikey!: GoogleMapsSectionProps["apikey"];
+  @Prop() startLocation!: GoogleMapsSectionProps["startLocation"];
+  @Prop({ default: 15 }) zoom!: GoogleMapsSectionProps["zoom"];
+  @Prop({ default: "en" }) language!: GoogleMapsSectionProps["language"];
+  @Prop() locations!: GoogleMapsSectionProps["locations"];
+  @Prop({ default: "Contact Us" })
+  buttonLabel!: GoogleMapsSectionProps["buttonLabel"];
+  @Prop({ required: true })
+  handleButtonClick!: GoogleMapsSectionProps["handleButtonClick"];
+
+  @Watch("selectedIndex")
+  handleSelectionChange(index: number, oldIndex: number) {
+    if (
+      this.map &&
+      this.selectedIndex !== null &&
+      this.locations &&
+      this.markers
+    ) {
+      const marker = this.markers[index];
+      marker.infoWindow.open(this.map, marker.marker);
+      this.map.panTo(this.locations[index].position);
+      this.markers[oldIndex].infoWindow.close();
+    }
+  }
+
+  private renderDescriptionBox(location: MapsLocation): Node {
```

Comment:
Please provide an optional render-prop that passes the location and the div that will be passed to google maps